arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt

```diff
@@ -469,59 +505,77 @@ class ArcGISArView : FrameLayout, DefaultLifecycleObserver, Scene.OnUpdateListen
      * This function currently assumes that the [Context] of this view is an instance of [Activity] to ensure that we can
      * request permissions. This may not always be the case and the handling of permission are under review.
      *
-     * @since 100.6.0
-     */
-    @SuppressLint("MissingPermission") // suppressed as function returns if permission hasn't been granted
-    fun startTracking() {
-        (context as? Activity)?.let { activity ->
-            // ARCore requires camera permissions to operate. If we did not yet obtain runtime
-            // permission on Android M and above, now is a good time to ask the user for it.
-            // when the permission is requested and the user responds to the request from the OS this is executed again
-            // during onResume()
-            if (isUsingARCore == ARCoreUsage.YES && renderVideoFeed && !hasPermission(
-                    CAMERA_PERMISSION
-                )
-            ) {
-                requestPermission(
-                    activity,
-                    CAMERA_PERMISSION,
-                    CAMERA_PERMISSION_CODE
-                )
-                return
-            }
+     * @throws [IllegalStateException] if the Context is not an instance of Activity
+     * @since 100.6.0
+     */
+    private fun startArCoreSession() {
+        // Throw exception if Context is not an instance of Activity as it's required for permission request
+        check(context is Activity) { "Context must be an instance of Activity" }
+
+        // ARCore requires camera permissions to operate. If we did not yet obtain runtime
+        // permission on Android M and above, now is a good time to ask the user for it.
+        // when the permission is requested and the user responds to the request from the OS this is executed again
+        // during onResume()
+        if (isUsingARCore == ARCoreUsage.YES && renderVideoFeed && !hasPermission(
+                CAMERA_PERMISSION
+            )
```

Comment:
Strange line break, can this all go on one line?

Reply:
Yeah, will fix.